Display an icon based on the state of a check

dmarcoux opened the pull request:
Closes #8741

Preview:
![issue-8741-2nd](https://user-images.githubusercontent.com/1102934/69716440-4993f580-110a-11ea-8005-a2dd9a5d5771.png)

It's not directly related to the issue, but I'm wondering if we should sort the checks by state... Any thoughts?

DavidKang commented:
@dmarcoux, what about if we replace the bullets with the icons?

coolo commented:
About your question: I think grouping by state makes sense as on the #index page we only show failed checks, having the same set on the #show page may make it easier to recognize.

dmarcoux commented:
> @dmarcoux, what about if we replace the bullets with the icons?

Here's a preview of how this would look. I removed the list `<ul>` has `padding-left 0; list-style: none`. I added some missing checks to have a better overview of the changes.

![issue-8741-david-suggestion](https://user-images.githubusercontent.com/1102934/69639380-b3ee5c80-105c-11ea-932b-ad2ea2eb8c70.png)

coolo commented:
now missing checks miss some bullet, no?

coolo commented:
(can we have a transparent icon there? :)

dmarcoux commented:
> now missing checks miss some bullet, no?

This is intended. It's always the same text for a missing check (beside the name of the check), so I felt like adding an icon doesn't bring value here.

> (can we have a transparent icon there? :)

Hum... what do you mean? Is it to align the missing checks with the checks?

coolo commented:
> Hum... what do you mean? Is it to align the missing checks with the checks?

right.

dmarcoux commented:
I just pushed a new set of changes and updated the screenshot in the PR's description. Let me know what you think.

coolo commented:
the icon for pending is a little larger than the rest, but I guess that's only noticable as your check names are very similiar. So I'm good

dmarcoux commented:
> the icon for pending is a little larger than the rest, but I guess that's only noticable as your check names are very similiar. So I'm good

I also noticed that when making sure this aligns well. It's exactly 2 pixels larger. I ended up leaving this as it is.
